shared/javascripts/context_messenger.js

```diff
@@ -471,12 +471,39 @@ if (Privly === undefined) {
   SafariAdapter.prototype.setListener = function (callback) {
     if (this.getContextName() === 'BACKGROUND_SCRIPT') {
       safari.application.addEventListener("message", function(payload) {
-        callback(payload);
+        if (payload.name !== "undefined" && payload.name === "privlyMessage") {
+          callback(payload.message);
+          payload = payload.message;
+        } else {
+          callback(payload);
+        }
+        // Update the options/changed values in the background page
+        if (payload.body !== "undefined" && payload.body.action === "options/changed") {
```

Comment:
So all these lines are to ensure the background script has the same local storage options as the privly-app? It would be better to change the local storage script to send/receive a storage message for the safari platform that will ensure the two scripting contexts store all the same values. This will be less brittle.